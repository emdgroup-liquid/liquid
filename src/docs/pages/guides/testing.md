---
eleventyNavigation:
  key: Testing
  parent: Guides
  order: 10
layout: layout.njk
title: Testing
permalink: guides/testing/
---

# Testing

If you want to __unit test__ code in your app, which depends on the functionality of Liquid Oxygen Web Components, without mocking their internals, then you must make sure all relevant components have _hydrated_ in your test setup, before you execute your test code. You can do so by running the `defineCustomElements()` method imported from `'@emdgroup-liquid/liquid/dist/loader'` before each test and a `await`ing a helper function, which makes sure all components involved in the test have hydrated.

The following example shows how to execute unit tests with hydrated Web Components with [vitest](https://vitest.dev/), React, and the [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/).

First, [configure vitest to use a setup file](https://vitest.dev/config/#setupfiles):

```js
// vite.config.ts
import { defineConfig } from 'vitest/config'
export default defineConfig({
  // ...
  test: {
    // ...
    setupFiles: './src/setupTests.ts',
  },
})
```

The `setupTests.ts` file looks like this:

```js
// src/setupTests.ts
import { defineCustomElements } from '@emdgroup-liquid/liquid/dist/loader'
import '@testing-library/jest-dom'
import matchers from '@testing-library/jest-dom/matchers'
import { cleanup } from '@testing-library/react'
import { vi } from 'vitest'

// Extend Vitest's expect method with methods from react-testing-library.
expect.extend(matchers)

beforeAll(() => {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  global.__LD_ASSET_PATH__ = '/' // removes CDN warning
  defineCustomElements()

  global.fetch = vi
    .fn()
    .mockReturnValue(Promise.resolve({ text: () => Promise.resolve('') }))
})

// Run a cleanup after each test case.
afterEach(() => {
  cleanup()
})
```

In your test file, you can now wait for the Web Components to hydrate before executing your actual test code.

```tsx
// src/App.test.tsx
import App from './App'
import { render, screen, waitFor, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'

async function waitForLiquidToHydrate() {
  return waitFor(() => {
    Array.from(document.querySelectorAll('*'))
      .filter((e) => e.tagName.startsWith('LD-'))
      .forEach((component) => {
        expect(component).toHaveClass('hydrated')
      })
  })
}

test('validates input', async () => {
  render(<App />)

  await waitForLiquidToHydrate()

  const ldInputName = screen.getByPlaceholderText('e.g. Jason Parse')
  expect(ldInputName).toBeInTheDocument()

  const ldButtonSubmit = screen.getByText('Submit')
  expect(ldButtonSubmit).toBeInTheDocument()
  userEvent.click(ldButtonSubmit)

  const form = await screen.findByTestId('form')
  expect(form).toBeInTheDocument()

  fireEvent.submit(form)

  await screen.findByText('Your full name is required.')
})
```

Check out our [React sandbox app](https://stackblitz.com/github/emdgroup-liquid/liquid-sandbox-react-tailwind?file=src%2FApp.test.tsx) for a full working example.

<docs-page-nav prev-href="guides/design-tokens/" next-title="Sandbox applications" next-href="guides/sandbox-applications/"></docs-page-nav>

