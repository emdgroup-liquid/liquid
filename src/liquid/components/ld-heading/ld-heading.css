.ld-heading--h1 {
  font: var(--ld-typo-h1);
}
.ld-heading--h2 {
  font: var(--ld-typo-h2);
}
.ld-heading--h3 {
  font: var(--ld-typo-h3);
}
.ld-heading--h4 {
  font: var(--ld-typo-h4);
}
.ld-heading--h5 {
  font: var(--ld-typo-h5);
}
.ld-heading--h6 {
  font: var(--ld-typo-h6);
}
.ld-heading--b1 {
  font: var(--ld-typo-b1);
  text-transform: uppercase;
}
.ld-heading--b2 {
  font: var(--ld-typo-b2);
  text-transform: uppercase;
}
.ld-heading--b3 {
  font: var(--ld-typo-b3);
  text-transform: uppercase;
}
.ld-heading--b4 {
  font: var(--ld-typo-b4);
  text-transform: uppercase;
}
.ld-heading--b5 {
  font: var(--ld-typo-b5);
  text-transform: uppercase;
}
.ld-heading--b6 {
  font: var(--ld-typo-b6);
  text-transform: uppercase;
}
.ld-heading--xb1 {
  font: var(--ld-typo-xb1);
  text-transform: uppercase;
}
.ld-heading--xb2 {
  font: var(--ld-typo-xb2);
  text-transform: uppercase;
}
.ld-heading--xb3 {
  font: var(--ld-typo-xb3);
  text-transform: uppercase;
}
.ld-heading--xh1 {
  font: var(--ld-typo-xh1);
}
.ld-heading--xh2 {
  font: var(--ld-typo-xh2);
}
.ld-heading--xh3 {
  font: var(--ld-typo-xh3);
}
.ld-heading--xh4 {
  font: var(--ld-typo-xh4);
}
.ld-heading--xh5 {
  font: var(--ld-typo-xh5);
}
.ld-heading--xh6 {
  font: var(--ld-typo-xh6);
}

.ld-heading--h1,
.ld-heading--h2,
.ld-heading--h3,
.ld-heading--h4,
.ld-heading--h5,
.ld-heading--h6,
.ld-heading--xb1,
.ld-heading--xb2,
.ld-heading--xb3,
.ld-heading--xh1,
.ld-heading--xh2,
.ld-heading--xh3,
.ld-heading--xh4,
.ld-heading--xh5,
.ld-heading--xh6 {
  font-weight: 700;
}

:where(ld-heading[visual-level='b1']),
:where(ld-heading[visual-level='b2']),
:where(ld-heading[visual-level='b3']),
:where(ld-heading[visual-level='b4']),
:where(ld-heading[visual-level='b5']),
:where(ld-heading[visual-level='b6']),
:where(ld-heading[visual-level='xb1']),
:where(ld-heading[visual-level='xb2']),
:where(ld-heading[visual-level='xb3']),
:where(:not(ld-heading) > .ld-heading--b1),
:where(:not(ld-heading) > .ld-heading--b2),
:where(:not(ld-heading) > .ld-heading--b3),
:where(:not(ld-heading) > .ld-heading--b4),
:where(:not(ld-heading) > .ld-heading--b5),
:where(:not(ld-heading) > .ld-heading--b6),
:where(:not(ld-heading) > .ld-heading--xb1),
:where(:not(ld-heading) > .ld-heading--xb2),
:where(:not(ld-heading) > .ld-heading--xb3) {
  color: var(--ld-col-rb-default);

  :where(.ld-theme-ocean) &,
  :where([class*='ld-theme'] .ld-theme-ocean) & {
    color: var(--ld-col-rb-default);
  }

  :where(.ld-theme-solvent) &,
  :where([class*='ld-theme'] .ld-theme-solvent) &,
  :where(.ld-theme-bubblegum) &,
  :where([class*='ld-theme'] .ld-theme-bubblegum) &,
  :where(.ld-theme-shake) &,
  :where([class*='ld-theme'] .ld-theme-shake) & {
    color: var(--ld-col-rp-default);
  }

  :where(.ld-theme-tea) &,
  :where([class*='ld-theme'] .ld-theme-tea) & {
    color: var(--ld-col-rg-default);
  }
}
