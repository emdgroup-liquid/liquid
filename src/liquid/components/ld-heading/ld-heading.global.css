ld-heading[visual-level*='b'],
.ld-heading--b1,
.ld-heading--b2,
.ld-heading--b3,
.ld-heading--b4,
.ld-heading--b5,
.ld-heading--b6,
.ld-heading--xb1,
.ld-heading--xb2,
.ld-heading--xb3 {
  .ld-theme-ocean :where(&),
  [class*='ld-theme'] .ld-theme-ocean :where(&) {
    color: var(--ld-col-rb-default);
  }

  .ld-theme-solvent :where(&),
  [class*='ld-theme'] .ld-theme-solvent :where(&),
  .ld-theme-bubblegum :where(&),
  [class*='ld-theme'] .ld-theme-bubblegum :where(&),
  .ld-theme-shake :where(&),
  [class*='ld-theme'] .ld-theme-shake :where(&) {
    color: var(--ld-col-rp-default);
  }

  .ld-theme-tea :where(&),
  [class*='ld-theme'] .ld-theme-tea :where(&) {
    color: var(--ld-col-rg-default);
  }
}
