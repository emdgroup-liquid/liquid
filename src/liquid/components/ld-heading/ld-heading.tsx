import { Component, Element, h, Prop } from '@stencil/core'
import { cloneAttributes } from '../../utils/cloneAttributes'
import { JSXBase } from '@stencil/core/internal'
import HeadingHTMLAttributes = JSXBase.HTMLAttributes
import { applyPropAliases } from '../../utils/applyPropAliases'

@Component({
  tag: 'ld-heading',
  styleUrl: 'ld-heading.css',
  shadow: false,
})
export class LdHeading {
  @Element() el: HTMLHeadingElement

  /** The heading level. */
  @Prop() level!: 1 | 2 | 3 | 4 | 5 | 6 | '1' | '2' | '3' | '4' | '5' | '6'

  /**
   * The heading style. Overrides the style inferred from the heading level.
   */
  @Prop({
    mutable: true,
  })
  visualLevel:
    | undefined
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6'
    | 'b1'
    | 'b2'
    | 'b3'
    | 'b4'
    | 'b5'
    | 'b6'
    | 'xb1'
    | 'xb2'
    | 'xb3'
    | 'xh1'
    | 'xh2'
    | 'xh3'
    | 'xh4'
    | 'xh5'
    | 'xh6'

  /**
   * Since b1 to b6 headings are uppercase headings, screen readers need to be served a
   * (non-uppercase) aria-label (otherwise they will read out the heading letter by letter).
   * If you're using a b1 to b6 or xb1 to xb3 visual level heading, an aria-label will be
   * set automatically on the heading element. The component will use the inner HTML for the
   * label implicitly. If you want to set an aria-label explicitly (such as when you have
   * inner HTML that should not be part of the label), you can use this property.
   */
  @Prop()
  ariaLabel: string

  private validateLevel() {
    if (![1, 2, 3, 4, 5, 6].includes(parseInt(this.level + '', 10))) {
      throw new TypeError(`ld-heading level prop invalid; got ${this.level}`)
    }
  }

  private validateVisualLevel() {
    if (
      ![
        undefined,
        'h1',
        'h2',
        'h3',
        'h4',
        'h5',
        'h6',
        'b1',
        'b2',
        'b3',
        'b4',
        'b5',
        'b6',
        'xb1',
        'xb2',
        'xb3',
        'xh1',
        'xh2',
        'xh3',
        'xh4',
        'xh5',
        'xh6',
      ].includes(this.visualLevel)
    ) {
      throw new TypeError(
        `ld-heading visualLevel prop invalid; got ${this.visualLevel}`
      )
    }
  }

  private applyAriaLabel() {
    const isBHeading =
      this.visualLevel?.indexOf('b') === 0 ||
      this.visualLevel?.indexOf('xb') === 0
    if (isBHeading) {
      const heading = this.el.querySelector('.ld-heading')
      heading.setAttribute(
        'aria-label',
        this.ariaLabel || heading.innerHTML.trim()
      )
    }
  }

  componentWillLoad() {
    applyPropAliases.apply(this)

    this.validateLevel()
    this.validateVisualLevel()
  }

  componentDidRender() {
    this.applyAriaLabel()
  }

  render() {
    const HTag = `h${this.level}`
    const cl = `ld-heading ld-heading--${this.visualLevel || 'h' + this.level}`

    return (
      <HTag
        class={cl}
        {...cloneAttributes<HeadingHTMLAttributes<HTMLHeadingElement>>(this.el)}
      >
        <slot></slot>
      </HTag>
    )
  }
}
