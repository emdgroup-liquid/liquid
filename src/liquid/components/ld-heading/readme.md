---
eleventyNavigation:
  key: Heading
  parent: Components
layout: layout.njk
title: Heading
permalink: components/ld-heading/
---

# ld-heading

<link rel="stylesheet" href="/dist/build/css/ld-heading.css">

Headings are used as an introduction into a topic or a content section and for visual and semantic differentiation between content blocks.

> **Note**: A common navigation technique for users of screen reading software is jumping from heading to heading to quickly determine the content of the page. Because of this, it is important to not skip one or more heading levels. Doing so may create confusion, as the person navigating this way may be left wondering where the missing heading is.
> 
> If you really must place headings in a way that they are visually not following a semantic hirarchicy, use the `visual-level` property for visually styling the heading according to the design given, while still applying correct semantical heading levels using the `level` prop. 

---

## Examples

### With heading level

{% example "html", true %}
<ld-heading level="1">
  Almost before we knew it, we had left the ground.
</ld-heading>

<ld-heading level="2">
  Almost before we knew it, we had left the ground.
</ld-heading>

<ld-heading level="3">
  Almost before we knew it, we had left the ground.
</ld-heading>

<ld-heading level="4">
  Almost before we knew it, we had left the ground.
</ld-heading>

<ld-heading level="5">
  Almost before we knew it, we had left the ground.
</ld-heading>

<ld-heading level="6">
  Almost before we knew it, we had left the ground.
</ld-heading>

<!-- CSS component -->

<h1 class="ld-heading ld-heading--h1">
  Almost before we knew it, we had left the ground.
</h1>

<h2 class="ld-heading ld-heading--h2">
  Almost before we knew it, we had left the ground.
</h2>

<h3 class="ld-heading ld-heading--h3">
  Almost before we knew it, we had left the ground.
</h3>

<h4 class="ld-heading ld-heading--h4">
  Almost before we knew it, we had left the ground.
</h4>

<h5 class="ld-heading ld-heading--h5">
  Almost before we knew it, we had left the ground.
</h5>

<h6 class="ld-heading ld-heading--h6">
  Almost before we knew it, we had left the ground.
</h6>
{% endexample %}

### With deviating visual level `b*`

`b*` headings should be only a couple of words long.

{% example "html", true %}
<ld-heading level="1" visual-level="b1">
  Lorem ipsum
</ld-heading>

<ld-heading level="2" visual-level="b2">
  Lorem ipsum
</ld-heading>

<ld-heading level="3" visual-level="b3">
  Lorem ipsum
</ld-heading>

<ld-heading level="4" visual-level="b4">
  Lorem ipsum
</ld-heading>

<ld-heading level="5" visual-level="b5">
  Lorem ipsum
</ld-heading>

<ld-heading level="6" visual-level="b6">
  Lorem ipsum
</ld-heading>

<!-- CSS component -->

<h1 class="ld-heading ld-heading--b1" aria-label="Lorem ipsum">
  Lorem ipsum
</h1>

<h2 class="ld-heading ld-heading--b2" aria-label="Lorem ipsum">
  Lorem ipsum
</h2>

<h3 class="ld-heading ld-heading--b3" aria-label="Lorem ipsum">
  Lorem ipsum
</h3>

<h4 class="ld-heading ld-heading--b4" aria-label="Lorem ipsum">
  Lorem ipsum
</h4>

<h5 class="ld-heading ld-heading--b5" aria-label="Lorem ipsum">
  Lorem ipsum
</h5>

<h6 class="ld-heading ld-heading--b6" aria-label="Lorem ipsum">
  Lorem ipsum
</h6>
{% endexample %}

### With deviating visual level `xh*`

{% example "html", true %}
<ld-heading level="1" visual-level="xh1">
  Almost before we knew it, we had left the ground.
</ld-heading>

<ld-heading level="2" visual-level="xh2">
  Almost before we knew it, we had left the ground.
</ld-heading>

<ld-heading level="3" visual-level="xh3">
  Almost before we knew it, we had left the ground.
</ld-heading>

<!-- CSS component -->

<h1 class="ld-heading ld-heading--xh1">
  Almost before we knew it, we had left the ground.
</h1>

<h2 class="ld-heading ld-heading--xh2">
  Almost before we knew it, we had left the ground.
</h2>

<h3 class="ld-heading ld-heading--xh3">
  Almost before we knew it, we had left the ground.
</h3>
{% endexample %}

### With deviating visual level `xb*`

`xb*` headings should be only a couple of words long.

{% example "html", true %}
<ld-heading level="1" visual-level="xb1">
  Lipsum
</ld-heading>

<ld-heading level="2" visual-level="xb2">
  Lipsum
</ld-heading>

<ld-heading level="3" visual-level="xb3">
  Lipsum
</ld-heading>

<!-- CSS component -->

<h1 class="ld-heading ld-heading--xb1" aria-label="Lorem ipsum">
  Lipsum
</h1>

<h2 class="ld-heading ld-heading--xb2" aria-label="Lorem ipsum">
  Lipsum
</h2>

<h3 class="ld-heading ld-heading--xb3" aria-label="Lorem ipsum">
  Lipsum
</h3>
{% endexample %}

## Heading colors

You may have noticed that all `b*` and `xb*` headings use a color given by the current theme, while all `h*` and `xh*` headings inherit their color. While this is the default behaviour, you can still apply a different color on both types of headings. This may be usefull, if, for instance, you need to display a `b1` heading in "vibrant yellow" on top of a background using a "rich color".

> **Note**: With great power comes great responsibility. Make sure you follow the brand guidelines. Especially make sure that the text remains readable at all times. Check the text for accessibility issues, such as [insufficiant contrast](https://www.w3.org/TR/WCAG21/#contrast-minimum).

Here are some examples on how you can apply different colors on headings:

{% example "html", true %}
<style>
.custom-color { color: var(--ld-col-vm-default); }
</style>

<ld-heading level="1" visual-level="b1" class="custom-color">
  Lorem ipsum
</ld-heading>

<ld-heading level="2" visual-level="h1" style="color: var(--ld-col-vc4)">
  Almost before we knew it, we had left the ground.
</ld-heading>

<!-- CSS component -->

<style>
.custom-color { color: var(--ld-col-vm-default); }
</style>

<h1 class="ld-heading ld-heading--b1 custom-color" aria-label="Lorem ipsum">
  Lorem ipsum
</h1>

<h2 class="ld-heading ld-heading--h1" style="color: var(--ld-col-vc4)">
  Almost before we knew it, we had left the ground.
</h2>

{% endexample %}

<!-- Auto Generated Below -->


## Properties

| Property             | Attribute      | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Type                                                                                                                                                                            | Default     |
| -------------------- | -------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `ariaLabel`          | `aria-label`   | Since b* and xb* headings are uppercase headings, screen readers need to be served a (non-uppercase) aria-label (otherwise they will read out the heading letter by letter). If you're using a b* or xb* visual level heading, an aria-label will be set automatically on the heading element. The component will use the inner HTML for the label implicitly. If you want to set an aria-label explicitly (such as when you have inner HTML that should not be part of the label), you can use this property. | `string`                                                                                                                                                                        | `undefined` |
| `key`                | `key`          | for tracking the node's identity when working with lists                                                                                                                                                                                                                                                                                                                                                                                                                                                       | `string \| number`                                                                                                                                                              | `undefined` |
| `level` _(required)_ | `level`        | The heading level.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | `"1" \| "2" \| "3" \| "4" \| "5" \| "6" \| 1 \| 2 \| 3 \| 4 \| 5 \| 6`                                                                                                          | `undefined` |
| `ref`                | `ref`          | reference to component                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | `any`                                                                                                                                                                           | `undefined` |
| `visualLevel`        | `visual-level` | The heading style. Overrides the style inferred from the heading level.                                                                                                                                                                                                                                                                                                                                                                                                                                        | `"h1" \| "h2" \| "h3" \| "h4" \| "h5" \| "h6" \| "b1" \| "b2" \| "b3" \| "b4" \| "b5" \| "b6" \| "xb1" \| "xb2" \| "xb3" \| "xh1" \| "xh2" \| "xh3" \| "xh4" \| "xh5" \| "xh6"` | `undefined` |


----------------------------------------------

 
