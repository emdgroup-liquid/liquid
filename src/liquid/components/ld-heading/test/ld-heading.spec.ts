import { newSpecPage } from '@stencil/core/testing'
import { LdHeading } from '../ld-heading'

describe('ld-heading', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [LdHeading],
      html: `<ld-heading level="1">Text</ld-heading>`,
    })
    expect(page.root).toMatchSnapshot()
  })
  it('throws if no level is provided', async () => {
    expect.assertions(1)
    try {
      await newSpecPage({
        components: [LdHeading],
        html: `<ld-heading>Text</ld-heading>`,
      })
    } catch (err) {
      expect(err).toStrictEqual(
        TypeError('ld-heading level prop invalid; got undefined')
      )
    }
  })
  it('throws if invalid level is provided', async () => {
    try {
      await newSpecPage({
        components: [LdHeading],
        html: `<ld-heading level="7">Text</ld-heading>`,
      })
      expect(true).toBe(false)
    } catch (err) {
      expect(err).toStrictEqual(
        TypeError('ld-heading level prop invalid; got 7')
      )
    }
  })
  it('renders with visual level', async () => {
    const page = await newSpecPage({
      components: [LdHeading],
      html: `<ld-heading level="1" visual-level="h3">Text</ld-heading>`,
    })
    expect(page.root).toMatchSnapshot()
  })
  it('renders with visual b level and explicit aria-label', async () => {
    const page = await newSpecPage({
      components: [LdHeading],
      html: `<ld-heading level="1" visual-level="b3" aria-label="Yolo">Text</ld-heading>`,
    })
    expect(page.root).toMatchSnapshot()
  })
  it('renders with visual b level and implicit aria-label', async () => {
    const page = await newSpecPage({
      components: [LdHeading],
      html: `<ld-heading level="1" visual-level="b3">Text</ld-heading>`,
    })
    expect(page.root).toMatchSnapshot()
  })
  it('throws with invalid visual level prop', async () => {
    let error: unknown

    try {
      await newSpecPage({
        components: [LdHeading],
        html: `<ld-heading level="1" visual-level="asdf" aria-label="Text">Text</ld-heading>`,
      })
    } catch (err) {
      error = err
    }

    expect(error).toStrictEqual(
      TypeError('ld-heading visualLevel prop invalid; got asdf')
    )
  })
})
