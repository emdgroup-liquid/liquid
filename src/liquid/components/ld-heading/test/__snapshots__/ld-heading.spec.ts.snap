// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ld-heading renders 1`] = `
<ld-heading level="1">
  <mock:shadow-root>
    <h1 class="ld-heading ld-heading--h1" part="tag">
      <slot></slot>
    </h1>
  </mock:shadow-root>
  Text
</ld-heading>
`;

exports[`ld-heading renders with visual b level and explicit aria-label 1`] = `
<ld-heading aria-label="Yolo" level="1" visual-level="b3">
  <mock:shadow-root>
    <h1 aria-label="Yolo" class="ld-heading ld-heading--b3" part="tag">
      <slot></slot>
    </h1>
  </mock:shadow-root>
  Text
</ld-heading>
`;

exports[`ld-heading renders with visual b level and implicit aria-label 1`] = `
<ld-heading level="1" visual-level="b3">
  <mock:shadow-root>
    <h1 aria-label="Text" class="ld-heading ld-heading--b3" part="tag">
      <slot></slot>
    </h1>
  </mock:shadow-root>
  Text
</ld-heading>
`;

exports[`ld-heading renders with visual level 1`] = `
<ld-heading level="1" visual-level="h3">
  <mock:shadow-root>
    <h1 class="ld-heading ld-heading--h3" part="tag">
      <slot></slot>
    </h1>
  </mock:shadow-root>
  Text
</ld-heading>
`;
